### PR TITLE
Add io.elementary.Loki.BaseApp

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
This is a work in progress. The BaseApp includes for now just the most important stuff: granite, libgee and the elementary Gtk theme. 
@codygarver Do you think you can help with providing the list of the packages that are required to build basic elementary Gtk Apps. This will allow elementary app to be published pretty easily on all the platforms using Flatpak.

### CODE REVIEW

**Comment by cassidyjames** [comment]
How does versioning work with this? Right now elementary OS has two target versions, Loki (0.4) and the upcoming Juno (5.0) release. Library versions, GTK quirks, etc. differ between the two. So an app built only for 0.4 should be provided with an 0.4 runtime if possible.

**Comment by bilelmoussaoui** [comment]
That's what I'm planning to do. To provide two different versions for the two releases with the correct supported versions of the used libraries. Loki is a priority for now.

**Comment by bilelmoussaoui** [comment]
Btw, I have renamed the BaseApp to io.elementary.Loki. The next release would be named io.elementary.Juno.

**Comment by nedrichards** [comment]
bot, build io.elementary.Loki

**Comment by bilelmoussaoui** [comment]
@nedrichards Shouldn't be `bot, build io.elementary.Loki.BaseApp` instead?

**Comment by nedrichards** [comment]
Yep, I can't type.

**Comment by nedrichards** [comment]
bot, build io.elementary.Loki.BaseApp

**Comment by bilelmoussaoui** [comment]
The build fails due to a missing Appdata file. Is it really needed for an AppBase?

**Comment by nedrichards** [comment]
No, you need a flathub.json with the following magic string: https://github.com/flathub/org.electronjs.Electron2.BaseApp/blob/master/flathub.json

**Comment by nedrichards** [comment]
bot, build io.elementary.Loki.BaseApp

**Comment by bilelmoussaoui** [comment]
The build fails due to a missing dependency for the latest vala version. I'm gonna check which version's shipped along with Loki. To avoid unneeded patches to build the apps as flatpak.

**Comment by nedrichards** [comment]
bot, build io.elementary.Loki.BaseApp

**Comment by bilelmoussaoui** [comment]
This should build correctly now. @nedrichards Can you trigger a new build, please?

**Comment by nedrichards** [comment]
bot, build io.elementary.Loki.BaseApp

File: io.elementary.Loki.BaseApp.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                "type": "archive",
+                "url": "https://download.gnome.org/sources/libgee/0.20/libgee-0.20.0.tar.xz",
+                "sha256": "21308ba3ed77646dda2e724c0e8d5a2f8d101fb05e078975a532d7887223c2bb"
+            }]
+        },
+        {
+            "name": "granite",
+            "buildsystem": "cmake-ninja",
+            "config-opts": [
+                "-DCMAKE_BUILD_TYPE=Release"
```
Comment (nedrichards) [inline io.elementary.Loki.BaseApp.json]: Is there a releasewithdebinfo option? The debug info should be stripped by flatpak into a .Debug extension but it'd likely be helpful for developers to have this enabled.

(note: this is a suggestion not a requirement)

**Comment by nedrichards** [comment]
bot, build io.elementary.Loki.BaseApp

**Comment by bilelmoussaoui** [comment]
Ready to go https://flathub.org/builds/#/builders/2/builds/2367

**Comment by bilelmoussaoui** [comment]
Once this is merged. I'm going to create a PR on every elementary app to use it. Also I'm going to push a bunch of new elementary apps to flathub. Stay tuned!

**Comment by nedrichards** [comment]
@bilelmoussaoui anyone else we should add to the repo?

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/io.elementary.Loki.BaseApp

**Comment by bilelmoussaoui** [comment]
I will let you know if there's anyone that should be added

**Pull request closed without merging**